test/IECoreScene/CurvesAlgoTest.py

```diff
@@ -470,6 +470,38 @@ def testBSplineCurvesIndexedFaceVaryingToVarying( self ) :
 		self.assertEqual( p.interpolation, IECoreScene.PrimitiveVariable.Interpolation.Varying )
 		self.assertEqual( p.data, IECore.FloatVectorData( range( 0, 3 ) ) )
 		self.assertEqual( p.indices, IECore.IntVectorData( [ 0, 1, 2, 0, 1, 2, 0, 1, 2, 0, 1, 2 ] ) )
+
+	def testBSplineCurvesPeriodic( self ) :
+		# Periodic curves are a corner case that we don't test much, but we should at least make sure that
+		# we produce valid primvars
+
+		curves = self.curvesBSpline()
+		for q in [ "d", "e", "h", "i", "varying_Color_V3f", "facevarying_Normal_V3f" ]:
+			del curves[q]
+
+		curves.setTopology( curves.verticesPerCurve(), curves.basis(), True )
+
+		p = IECoreScene.PrimitiveVariable( IECoreScene.PrimitiveVariable.Interpolation.Uniform, curves["c"].data )
+		IECoreScene.CurvesAlgo.resamplePrimitiveVariable(curves, p, IECoreScene.PrimitiveVariable.Interpolation.Varying )
+
+		self.assertTrue( curves.isPrimitiveVariableValid( p ) )
+
+		IECoreScene.CurvesAlgo.resamplePrimitiveVariable(curves, p, IECoreScene.PrimitiveVariable.Interpolation.Uniform )
+		self.assertEqual( curves["c"], p )
+
+		p = IECoreScene.PrimitiveVariable( IECoreScene.PrimitiveVariable.Interpolation.Vertex, IECore.FloatVectorData( [ i for i in range( curves.variableSize( IECoreScene.PrimitiveVariable.Interpolation.Vertex ) ) ] ) )
+
+		# Work around weird limitation that for resampling Varying -> Vertex the primvar must be stored on curves
+		curves["dummy"] = p
```

Comment:
I briefly tried to figure out why this might be necessary, but failed. Could you educate me?